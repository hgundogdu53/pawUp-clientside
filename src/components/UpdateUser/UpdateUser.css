.flex-container {
    margin-top: 1.5rem;
}

.update-user-section {
    transition: 0.3s;
    border-radius: 5px;
    
    width: 22.5rem;
    margin-top: 0.5rem;
    background-color: rgba(202,244,241, 0.2);
    width: 30rem;
}

.update-user-section:hover {
    box-shadow: 0 8px 16px 0 rgba(255, 255, 255, 0.151);
}

.sub-btn,
.res-btn {
    background-color: #fff;
    font-weight: bold;
    text-decoration: none;
    margin-top: 15px;
    padding: 7.5px 20px;
    text-decoration: none;
    display: inline-block;
    transition: all .2s;
    border-radius: 100px;
}

.sub-btn:hover, .res-btn:hover {
    cursor: pointer;
    transform: translateY(-2.5px);
}

.sub-btn:active, .res-btn:hover {
    transform: translateY(-1px);
}

.update-user-form {
    padding: 1rem;
}

.update-user-form label, .update-user-form input {
    display: block;
    width: 100%;
    text-align: start;
    margin-top: 0.5rem;
}

.update-user-form input {
    border: 1px solid #ccc;
    padding: 0.2rem 0.5rem;
    font: inherit;
}

.update-user-form input:focus {
    outline: none;
    border-color: rgb(4, 127, 168);
}

.update-user-form label {
    font-weight: bold;
}


.update-user-section {
    padding: 5px 75px;
}

.update-user-title {
    text-align: center;
    font-size: 1.75rem;
}

.update-user-form button[type="submit"],
.update-user-form button[type="reset"] {
    display: inline-block;
    margin-top: 1rem;
}


@media only screen and (min-width: 40rem) {
    .signup_form {
        margin: auto;
        width: 35rem;
    }
}

@media only screen and (min-width: 782px) {
    .update-user-section {
        width: 40rem;
    }
}
