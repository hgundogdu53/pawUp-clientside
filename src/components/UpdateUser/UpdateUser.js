import React, { Component } from 'react';
import TokenService from '../../services/token-service';
import config from '../../config';
import './UpdateUser.css';


class UpdateUser extends Component {
    constructor(props) {
        super(props);
        this.state = {
            users: []
        }
    }

    componentDidMount() {
        const options = {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': `bearer ${TokenService.getAuthToken()}`
            }
        }

        fetch(`${config.API_ENDPOINT}/users/user`, options)
            .then(res => {
                if (res.ok) {
                    return res.json();
                }
                else {
                    throw new Error('Something went wrong');
                }
            })
            .then(data => {
                this.setState({ users: data })
            })
            .catch(err => {
                throw new Error(err)
            }
            )
    }

    updateUser = e => {
        e.preventDefault();

        const inputVal = {
            first_name: e.target.first_name.value,
            last_name: e.target.last_name.value,
            email: e.target.email.value,
        }

        fetch(`${config.API_ENDPOINT}/users/user`, {
            method: 'PATCH',
            body: JSON.stringify(inputVal),
            headers: {
                "Content-Type": "application/json",
                "Authorization": `bearer ${TokenService.getAuthToken()}`
            }
        })
            .then(res => {
                if (res.ok) {
                    TokenService.clearAuthToken();
                    window.location.href = '/login'
                } else {
                    return res.json().then(error => {
                        throw error;
                    });
                }
            })
    }

    render() {
        if (this.state.users) {
            return (
                <main className='flex-container'>
                    <header>
                        <h1 className='update-user-title'>Update User</h1>
                    </header>
                    <section className='update-user-section'>
                        <form id="update-user-form" className='update-user-form' onSubmit={this.updateUser}>
                            <div className="form-section">
                                <label htmlFor="first_name">First Name</label>
                                <input type="text" name="first_name" id="first_name" defaultValue={this.state.users.first_name} required />
                            </div>
                            <div className="form-section">
                                <label htmlFor="last_name">Last Name</label>
                                <input type="text" name="last_name" id="last_name" defaultValue={this.state.users.last_name} required />
                            </div>
                            <div className="form-section">
                                <label htmlFor="email">Email</label>
                                <input type="email" name="email" id="email" defaultValue={this.state.users.email} required />
                            </div>
                            <button type="submit" className='sub-btn'>Submit</button>
                            <button type="reset" className='res-btn'>Reset</button>
                        </form>
                    </section>
                </main>
            )
        } else {
            return (
                <main>
                    <h1>User Does Not Exist!</h1>
                    <h2>Try Again</h2>
                </main>
            )
        }
    }
}

export default UpdateUser;