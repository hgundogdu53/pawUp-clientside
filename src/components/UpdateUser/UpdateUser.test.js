import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UpdateUser from './UpdateUser';

describe(`UpdateUser component`, () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<UpdateUser />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});