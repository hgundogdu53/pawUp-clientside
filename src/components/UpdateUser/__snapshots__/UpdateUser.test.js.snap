// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateUser component renders without crashing 1`] = `
<main
  className="flex-container"
>
  <header>
    <h1
      className="update-user-title"
    >
      Update User
    </h1>
  </header>
  <section
    className="update-user-section"
  >
    <form
      className="update-user-form"
      id="update-user-form"
      onSubmit={[Function]}
    >
      <div
        className="form-section"
      >
        <label
          htmlFor="first_name"
        >
          First Name
        </label>
        <input
          id="first_name"
          name="first_name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-section"
      >
        <label
          htmlFor="last_name"
        >
          Last Name
        </label>
        <input
          id="last_name"
          name="last_name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-section"
      >
        <label
          htmlFor="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          required={true}
          type="email"
        />
      </div>
      <button
        className="sub-btn"
        type="submit"
      >
        Submit
      </button>
      <button
        className="res-btn"
        type="reset"
      >
        Reset
      </button>
    </form>
  </section>
</main>
`;
